File: src/main/java/play/pluv/feed/domain/repository/FeedBookmarkRepository.java
package play.pluv.feed.domain.repository;

import java.util.List;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.data.jpa.repository.Query;
import play.pluv.feed.domain.FeedBookmark;

public interface FeedBookmarkRepository extends JpaRepository<FeedBookmark, Long> {

  @Query("""
      select fb
      from FeedBookmark fb
      join fetch fb.feed
      where fb.memberId = :memberId
      """)
  List<FeedBookmark> findByMemberIdWithJoin(final Long memberId);
}

File: src/main/kotlin/play/pluv/feed/domain/repository/FeedBookmarkRepository.kt
package play.pluv.feed.domain.repository

import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.data.jpa.repository.Query
import play.pluv.feed.domain.Feed
import play.pluv.feed.domain.FeedBookmark

interface FeedBookmarkRepository : JpaRepository<FeedBookmark, Long> {

    @Query(
        """
    select fb
    from FeedBookmark fb
    join fetch fb.feed
    where fb.memberId = :memberId
    """
    )
    fun findByMemberIdWithJoin(memberId: Long): List<FeedBookmark>

    fun existsByMemberIdAndFeed(memberId: Long, feed: Feed): Boolean

    fun deleteByMemberIdAndFeedId(memberId: Long, feedId: Long)
}
